src/helpers/config.ts

```diff
@@ -26,10 +27,11 @@ export interface AliasesNameMap {
 
 export async function checkWait(
   config: Config & Partial<AutoWaitConfig>,
-  receipt: WaitableTransactionReceipt
+  receipt: WaitableTransactionReceipt,
+  controller?: PollingController
 ): Promise<WaitableTransactionReceipt> {
   if (config.autoWait ?? false) {
-    const waited = await receipt.wait();
+    const waited = await receipt.wait(controller);
```

Comment:
This is the actual fix for the polling handling. `wait` was being called without any options / controller.